sip/sip-007-stacking-consensus.md

```diff
@@ -211,6 +212,62 @@ cycles are fixed length --- if fewer addresses participate in the
 Stacking rewards than there are slots in the cycle, then for the
 remaining blocks, all miners must send funds to burn addresses.
 
+## Adjusting Reward Threshold Based on Participation
+
+Each reward cycle may transfer miner funds to up to 5000 Bitcoin
+addresses. To ensure that this number of addresses is sufficient to
+cover the pool of participants (given 100% participation of liquid
+STX), the threshold for participation must be 0.02% (1/5000th) of the
+liquid supply of STX. However, if participation is _lower_ than 100%,
+the reward pool could admit lower STX holders. The Stacking protocol
+specifies **4 threshold operating levels**:
+
+* **25%** If fewer than `0.25 * STX_LIQUID_SUPPLY` STX participate in
+  a reward cycle, participant wallets controlling `x` STX may include
+  `floor(x / (0.00005*STX_LIQUID_SUPPLY))` addresses in the reward set.
+  (i.e., the minimum participation threshold is 1/20,000th of the liquid
+  supply).
+* **50%** If between `0.25 * STX_LIQUID_SUPPLY` and `0.5 * STX_LIQUID_SUPPLY`
+  STX participate in a reward cycle, participant wallets controlling `x`
+  STX may include `floor(x / (0.00010*STX_LIQUID_SUPPLY))` addresses in
+  the reward set. (i.e., the minimum participation threshold is 1/10,000th
+  of the liquid supply).
+* **75%** If between `0.5 * STX_LIQUID_SUPPLY` and `0.75 * STX_LIQUID_SUPPLY`
+  STX participate in a reward cycle, participant wallets controlling `x`
+  STX may include `floor(x / (0.00015*STX_LIQUID_SUPPLY))` addresses in
+  the reward set. (i.e., the minimum participation threshold is 3/20,000th
+  of the liquid supply).
+* **75%** If more than `0.75 * STX_LIQUID_SUPPLY` STX participate in a
+  reward cycle, participant wallets controlling `x` STX may include
+  `floor(x / (0.0002*STX_LIQUID_SUPPLY))` addresses in the reward
+  set. (i.e., the minimum participation threshold is 1/5,000th of the
+  liquid supply).
+
+In the event that a Stacker signals and locks up enough STX to submit
+multiple reward addresses, but only submits one reward address, that
+reward address will be included in the reward set multiple times.
+
+## Submitting Reward Address and Chain Tip Signaling
+
+Stacking participants must broadcast signed messages for three purposes:
+
+1. Indicating to the network how many STX should be locked up, and for
+   how many reward cycles.
+2. Indicate support for a particular chain tip.
+3. Specifying the Bitcoin address for receiving Stacking rewards.
+
+These messages may be broadcast either on the Stacks chain or the
+Bitcoin chain. If broadcast on the Stacks chain, these messages must
+be included in the Stacks chain _before_ the anchor block for the
+reward period. If broadcast on the Bitcoin chain, they may be
```

Comment:
Recommend adding something like: "Moreover, if sent to the Stacks chain, the anchor block must _confirm_ the transaction bearing the signed message."

I don't think this is true if the transaction is sent on the Bitcoin chain.  Sending a transaction on the Bitcoin chain effectively registers the chain tip signal in all subsequently-mined Stacks forks on that Bitcoin fork.

Reply:
The sentence "If broadcast on the Stacks chain, these messages must ￼be included in the Stacks chain _before_ the anchor block for the ￼reward period" is confusing to me.  Does it mean that if I signal on the Stacks chain, I have to get my transaction in before the prepare phase for the next reward cycle starts?  i.e. I have to get it in during the previous reward cycle, in order to be included in the next reward cycle?

Reply:
Yes -- the anchor block is the _latest possible_ view of Stacks data for the current reward cycle. That means that a stacking-message on the Stacks chain would need to be confirmed before the prepare phase begins.

Reply:
> I don't think this is true if the transaction is sent on the Bitcoin chain. Sending a transaction on the Bitcoin chain effectively registers the chain tip signal in all subsequently-mined Stacks forks on that Bitcoin fork.

True, but the message still "picks" a chain tip. If a message signals support for a chain tip that ends up not being the anchor block, should they qualify for Stacking rewards?

Reply:
No, they shouldn't receive rewards in that case.  But, just based on the text here, I don't know whether or not it's the case that the anchor block confirms any Bitcoin-based signaling transactions.